File: dcp-solutions/2024-11-18/solution.js
```javascript
function largestProduct(nums) {
    let sortedNums = [...nums].sort((a, b) => a - b);

    // Return the product of three largest numbers and the product of two smallest (most negative) numbers
    return Math.max(sortedNums[0] * sortedNums[1] * sortedNums[sortedNums.length - 1], 
                   sortedNums[0] * sortedNums[sortedNums.length - 1] * sortedNums[sortedNums.length - 2]);
}
```

File: dcp-solutions/2024-11-18/solution.go
```go
package main

import (
    "sort"
)

func largestProduct(nums []int) int {
    sort.Ints(nums)
    return max(nums[0]*nums[1]*nums[n-1], nums[n-2]*nums[n-3]*nums[-1])
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}
```